Feature/add openssl scripts

timhansendev opened the pull request:
Creates a Root CA and server cert for the nodejs https module for development.
The script is flexible to change many parameters via the bash environment, but sane defaults are provided.
Added ./src/certs folder for organization.
Update app.js to point to appropriate location for generated certs.
Update README.md on script for development purposes.

cappy1337 committed:
Add a directory for certificates.
Update app.js to the new locations, use absolute path.
Add tools directory.
Add script for generating a rootca + client certs for the app.

Signed-off-by: tim hansen <timhansendev@gmail.com>

cappy1337 committed:
Update README to include how to run and use the utility cert script.

Signed-off-by: tim hansen <timhansendev@gmail.com>

Aryuko commented:
It seems to dump all the files into the `/tools` directory for me and doesn't create any `/certs` directory at all. This means the `certsPath` you've defined doesn't work for me, and I assume this isn't intended. I tested this on macOS 10.15.3 using zsh.

Also, it doesn't work at all in PowerShell but I assume nobody cares about Windows so whatever 😂

timhansendev commented on tools/createCA.sh:
```diff
@@ -0,0 +1,109 @@
+	createClientKey
+	createClientCSR
+	acceptCSR
+	verifyCert
```
add `installRootCA` call here

timhansendev commented on tools/createCA.sh:
```diff
@@ -0,0 +1,109 @@
+# Install into certs folder
+installRootCA(){
+	mkdir -p ../src/certs/
+	cp "$client_key" ../src/certs/
```
change this to `mv` instead of `cp`

cappy1337 committed:
Fix comments.
Fix missing install function call.
Fix install function to mv instead of cp.
chmod so any user can utilize certificates.

Signed-off-by: tim hansen <timhansendev@gmail.com>

timhansendev commented:
Sorry about that Aryu, just a few silly mistakes. Please try again and let me know. I was simply missing a call to the function to install the certs into src/certs, as well as setting some ownership bits on the file.

cappy1337 committed:
Remove double join, allow user to specify in config.

Signed-off-by: tim hansen <timhansendev@gmail.com>

cappy1337 committed:
Merge branch 'master' of https://github.com/Aryuko/spotify-now-playing into feature/add_openssl_scripts

cappy1337 committed:
Add npm script for generating dev certs.

Signed-off-by: tim hansen <timhansendev@gmail.com>

cappy1337 committed:
Update readme

Signed-off-by: tim hansen <timhansendev@gmail.com>

cappy1337 committed:
Use BSD-compat commands for chmod to support OSX.

Signed-off-by: tim hansen <timhansendev@gmail.com>

cappy1337 committed:
Support only running from root directory of repo.
Fix relative paths for the certs.

Signed-off-by: tim hansen <timhansendev@gmail.com>

timhansendev commented:
@Aryuko My mistake on the npm `cert` script target- I left the relative paths `../src/certs` so it was going one directory above the root of the repo.  Latest push fixes thing since `npm run-script cert` always runs in the root path of the directory.

bradamson commented:
> I assume nobody cares about Windows

D:
